src/main/java/net/sf/jabref/logic/groups/GroupTreeNode.java

```diff
@@ -55,10 +57,37 @@ public AbstractGroup getGroup() {
     /**
      * Associates the specified group with this node.
      *
-     * @param group the new group (has to be non-null)
+     * @param newGroup the new group (has to be non-null)
      */
-    public void setGroup(AbstractGroup group) {
-        this.group = Objects.requireNonNull(group);
+    @Deprecated // use other overload
+    public void setGroup(AbstractGroup newGroup) {
+        this.group = Objects.requireNonNull(newGroup);
+    }
+
+    /**
+     * Associates the specified group with this node while also providing the possibility to modify previous matched
+     * entries so that they are now matched by the new group.
+     *
+     * @param newGroup the new group (has to be non-null)
+     * @param shouldKeepPreviousAssignments specifies whether previous matched entries should be carried over
+     * @param entriesInDatabase list of entries in the database
+     */
+    public Optional<EntriesGroupChange> setGroup(AbstractGroup newGroup, boolean shouldKeepPreviousAssignments,
+            List<BibEntry> entriesInDatabase) {
+        AbstractGroup oldGroup = getGroup();
+        setGroup(newGroup);
```

Comment:
Once the other method is no longer used it can be inlined (or at least made private).